scripts/create_topological_sort_of_all_services.py

```diff
@@ -0,0 +1,133 @@
+# coding: utf-8
+#
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This script generates topological sort of all the services based on how
+services are dependent on each other.
+"""
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import collections
+import os
+import sys
+import python_utils
+
+_PARENT_DIR = os.path.abspath(os.path.join(os.getcwd(), os.pardir))
+
+_PATHS_TO_INSERT = [
+    os.path.join(_PARENT_DIR, 'oppia_tools', 'esprima-4.0.1'),
+]
+
+for path in _PATHS_TO_INSERT:
+    sys.path.insert(0, path)
+
+# pylint: disable=wrong-import-position
+import esprima # isort:skip
+# pylint: enable=wrong-import-position
+
+NODES = set()
+VISIT_STACK = []
+TOPO_SORT_STACK = []
+ADJ_LIST = collections.defaultdict(list)
+DIRECTORY_NAMES = ['core/templates/dev/head', 'extensions']
+
+
+def dfs(node):
+    """Depth First Search starting with node.
+
+    Args:
+        node: str. The service name from which dfs will begin.
+    """
+    VISIT_STACK.append(node)
+    for pt in ADJ_LIST[node]:
+        if pt not in VISIT_STACK:
+            dfs(pt)
+    TOPO_SORT_STACK.append(node)
+
+
+def make_graph():
+    """Creates an adjaceny list considering services as node and dependencies
+    as edges.
+    """
+    for dirname in DIRECTORY_NAMES:
+        for root, _, filenames in os.walk(dirname):
+            for filename in filenames:
+                if filename.endswith((
+                        '.service.ts', 'Service.ts', 'Factory.ts')):
```

Comment:
Put these at the top of the file as a global variable since we want to generalize this.

Reply:
Done